packages/pwa/app/utils/utils.js

```diff
@@ -162,3 +164,107 @@ export const capitalize = (text) => {
  * @returns {object} - url object from the pwa-kit-config.json file
  */
 export const getUrlConfig = () => pwaKitConfig?.app?.url
+/**
+ * A util to return current multi site configuration
+ * @returns {object} - a list of site objects from the pwa-kit-config.json file
+ */
+export const getSitesConfig = () => pwaKitConfig?.app?.sites
+/**
+ * A util to return current defaultSiteId
+ * @returns {string} - default site Id string from pwa-kit-config file
+ */
+export const getDefaultSiteId = () => pwaKitConfig?.app?.defaultSiteId
+
+/**
+ * A util to get the siteId
+ * @param url
+ * @returns {string}
+ */
+export const getSiteId = (url) => {
+    let path = url
+    if (!path) {
+        path = `${window?.location.pathname}${window?.location.search}`
+    }
+
+    const {hostname, pathname, search} = new URL(`${getAppOrigin()}${path}`)
+    let siteId = getSiteIdByHostname(hostname)
+
+    if (siteId) {
+        return siteId
+    }
+
+    siteId = getSiteIdByAlias(`${pathname}${search}`)
+    return siteId
+}
+
+/**
+ * Get the site Id based on the given hostname
+ * @param {string} hostname
+ * @returns {string} siteId
+ */
+export const getSiteIdByHostname = (hostname) => {
+    const sitesConfig = getSitesConfig()
+    if (!sitesConfig.length) throw new Error('No site config found. Please check you configuration')
+    if (!hostname) return undefined
+
+    const site = sitesConfig.filter((site) => {
+        return site?.hostname?.some((i) => i.includes(hostname))
+    })
+
+    return site?.length === 1 ? site[0].id : undefined
+}
+
+/**
+ * get the site id based on the site alias in the given url
+ * @param {string} url - input url
+ * @returns {string} siteId
+ */
+export const getSiteIdByAlias = (url) => {
+    const [pathname, search] = url.split('?')
+
+    const defaultSiteId = getDefaultSiteId()
+    const sitesConfig = getSitesConfig()
+    const urlConfig = getUrlConfig()
```

Comment:
Continuing the idea about having a more generic config utility you could imagine something like this:

```
const {defaultSiteId, sites, url} = getConfig() 
```